docs/data/data-grid/editing/editing.md

```diff
@@ -401,6 +401,56 @@ The following demo requires a value for the **Payment method** column only if th
 
 {{"demo": "ConditionalValidationGrid.js", "disableAd": true, "bg": "inline", "defaultCodeOpen": false}}
 
+### Linked fields
+
+The options available for one field may depend on the value of another field.
+For instance, if the `singleSelect` column is used, you can provide a function to `valueOptions` returning the relevant options for the value selected in another field, as exemplified below.
+
+```tsx
+const columns: GridColDef[] = [
+  {
+    field: 'account',
+    type: 'singleSelect',
+    valueOptions: ({ row }) => {
+      return row.type === 'Income' // Gets the value of the "type" field
+        ? ['Sales', 'Investments', 'Ads']
+        : ['Taxes', 'Payroll', 'Utilities'];
+    },
+  },
+];
+```
+
+The code above is already enough to display different options in the **Account** column based on the value selected in the **Type** column.
+There is one task left which is to reset the account once the type is changed.
+This is needed because, once the type is changed, the previously selected account will not exist in the options.
+To solve that, you can create a custom edit component, reusing the built-in one, and pass a function to the `onChange` prop.
+This function should call `apiRef.current.setEditCellValue` to reset the value of the other field.
+
+```tsx
+const CustomTypeEditComponent = (props: GridEditSingleSelectCellProps) => {
+  const apiRef = useGridApiContext();
+
+  const handleChange = async () => {
+    await apiRef.current.setEditCellValue({
```

Comment:
I don't know how to improve this
But when reading it for the 1st time, I thought this `onChange` callback was switching the component to a control mode, meaning that users would also have to manually update the value of the current column.

I guess if I understood wrong, other will as well.

Reply:
Renaming it to `onValueChange` would be better?

Reply:
Maybe yes
@cherniavskii what's your opinion on this ?

Reply:
@flaviendelangle hmm, I didn't think about controlled prop when reading this, but I get what you mean.
I think `onValueChange` should be better then.

Reply:
Renamed to `onValueChange`.